frame/support/src/dispatch.rs

```diff
@@ -1325,13 +1331,27 @@ macro_rules! decl_module {
 				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_runtime_upgrade"));
 				let result: $return = (|| { $( $impl )* })();
 
-				$crate::crate_to_pallet_version!()
+				let new_storage_version = $crate::crate_to_pallet_version!();
+				new_storage_version
 					.put_into_storage::<<$trait_instance as $system::Config>::PalletInfo, Self>();
 
 				let additional_write = <
 					<$trait_instance as $system::Config>::DbWeight as $crate::traits::Get<_>
 				>::get().writes(1);
 
+				let pallet_name = <<
+					$trait_instance
+					as
+					$system::Config
+				>::PalletInfo as $crate::traits::PalletInfo>::name::<Self>().expect("pallet will have name in the runtime.");
+
+				$crate::debug::info!(
+					target: $crate::LOG_TARGET,
+					"⚠️ running migration for {} and setting new storage version to {:?}",
```

Comment:
And you want to print this for each runtime upgrade?

Reply:
exactly, see example in the description, I think it is very important to have an overview of wtf is doing what.